mijia/src/bluetooth.rs

```diff
@@ -0,0 +1,201 @@
+use crate::DBUS_METHOD_CALL_TIMEOUT;
+use anyhow::Context;
+use bluez_generated::generated::{OrgBluezAdapter1, OrgBluezDevice1};
+use core::fmt::Debug;
+use core::future::Future;
+use dbus::arg::cast;
+use dbus::arg::RefArg;
+use dbus::nonblock::stdintf::org_freedesktop_dbus::ObjectManager;
+use dbus::nonblock::SyncConnection;
+use futures::FutureExt;
+use itertools::Itertools;
+use std::collections::HashMap;
+use std::error::Error;
+use std::fmt::{self, Display, Formatter};
+use std::str::FromStr;
+use std::sync::Arc;
+
+#[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct DeviceId {
```

Comment:
probably wants a doc comment with the word opaque in it somewhere.

(observation) I wonder if id is the right name. I'm not used to "id"s being opaque structs with private fields. You have more experience of languages with private fields than me, so I will defer to your judgement on this. I guess we can't use Handle because we're already using that for JoinHandles and it would become confusing.

Reply:
Added a doc comment. Not sure what the best name is, I agree that it's a bit weird for something opaque to be called an ID.

Reply:
It turns out that TypeId [from the standard library](https://doc.rust-lang.org/src/core/any.rs.html#417-419) is defined as:
```
#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
#[stable(feature = "rust1", since = "1.0.0")]
pub struct TypeId {
    t: u64,
}
```
so I think we're probably doing the rustic thing here.